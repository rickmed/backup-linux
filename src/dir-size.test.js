const {dirSizeImp} = require('./dir-size')

const path = `/some/path`

it('returns a dir size in bytes', () => {
  const _execSync = jest.fn(() => `323213    /some/path`)
  const act = dirSizeImp(_execSync)(path)
  const exp = 323213000
  expect(act).toBe(exp)
})

it('calls execSync with the correct args', () => {
  const _execSync = jest.fn(() => `test str`)
  dirSizeImp(_execSync)(path)
  const act = _execSync.mock.calls[0]
  const exp = [
    `du ${path} -s --apparent-size`,
    {encoding: 'utf8'}
  ]
  expect(act).toEqual(exp)
})



