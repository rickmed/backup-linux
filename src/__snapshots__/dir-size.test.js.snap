exports[`test calls execSync with the correct args 1`] = `
Array [
  "du",
  Array [
    "--exclude=/dev/*",
    "--exclude=/proc/*",
    "--exclude=/sys/*",
    "--exclude=/tmp/*",
    "--exclude=/run/*",
    "--exclude=/mnt/*",
    "--exclude=/media/*",
    "--exclude=/var/run/*",
    "--exclude=/var/lock/*",
    "--exclude=/var/cache/apt/archives/*",
    "--exclude=/lost+found",
    "--exclude=/home/*/.gvfs",
    "--exclude=/home/*/.cache",
    "--exclude=/home/*/.local/share/Trash",
    "--exclude=/home/*/.thumbnails/*",
    "/some/path",
    "-s",
    "--apparent-size",
  ],
  Object {
    "encoding": "utf8",
  },
]
`;
